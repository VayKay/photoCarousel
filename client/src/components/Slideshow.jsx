/* eslint-disable class-methods-use-this */
/* eslint-disable react/jsx-one-expression-per-line */
/* eslint-disable react/button-has-type */
/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
import React from 'react';

class Slideshow extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      currIndex: this.props.modalFocus,
      translationAmount: -34,
    };

    this.nextPictureHandler = this.nextPictureHandler.bind(this);
    this.prevPictureHandler = this.prevPictureHandler.bind(this);
    this.logKey = this.logKey.bind(this);
  }

  componentDidMount() {
    const { currIndex, translationAmount } = this.state;
    let tmp = translationAmount;

    window.addEventListener('keydown', this.logKey);

    switch (currIndex) {
      case 0:
        tmp = 8;
        break;
      case 1:
        tmp = 8;
        break;
      case 2:
        tmp = -34;
        break;
      default:
        tmp -= ((currIndex - 2) * 80);
        break;
    }

    if (tmp !== translationAmount) {
      this.setState({
        translationAmount: tmp,
      });
    }
  }

  componentWillUnmount() {
    window.removeEventListener('keydown', this.logKey);
  }

  logKey({ code }) {
    if (code === 'ArrowRight') {
      this.nextPictureHandler();
    } else if (code === 'ArrowLeft') {
      this.prevPictureHandler();
    }
  }

  nextPictureHandler() {
    const { currIndex, translationAmount } = this.state;
    const { photos } = this.props;
    const tmp = currIndex + 1;
    let amt = translationAmount - 80;

    if (currIndex === 0 || currIndex === 1 || currIndex === photos.length - 1) {
      amt = 8;
    }

    if (currIndex === 2) {
      amt = -34;
    }

    if (currIndex !== photos.length - 1) {
      this.setState({
        currIndex: tmp,
        translationAmount: amt,
      });
    } else {
      this.setState({
        currIndex: 0,
        translationAmount: amt,
      });
    }
  }

  prevPictureHandler() {
    const { currIndex, translationAmount } = this.state;
    const { photos } = this.props;
    const tmp = currIndex - 1;
    let amt = translationAmount + 80;

    if (currIndex === 0) {
      amt = -34 - (photos.length - 4) * 80;
    }

    if (currIndex === 1 || currIndex === 2 || currIndex === 3) {
      amt = 8;
    }

    if (currIndex !== 0) {
      this.setState({
        currIndex: tmp,
        translationAmount: amt,
      });
    } else {
      this.setState({
        currIndex: photos.length - 1,
        translationAmount: amt,
      });
    }
  }

  render() {
    const { exitModal, modalFocus, photos } = this.props;
    const { currIndex, translationAmount } = this.state;
    return (
      <div>
        <div>
          <div dir="ltr">
            <div role="dialog" className="modal" style={{ zIndex: 2000 }}>
              <div data-veloute="slideshow-modal" className="slideshowModal">
                <div className="slideshowElements">
                  <div className="slideshowContainer">
                    <div>
                      <div>
                        <div className="exitButton">
                          <button type="button" className="hiddenButton" aria-busy="false" style={{ padding: 32, margin: -32 }} onClick={exitModal}>
                            <svg
                              viewBox="0 0 24 24"
                              role="img"
                              aria-label="Close"
                              focusable="false"
                              style={{
                                height: 24, width: 24, display: 'block', fill: 'rgb(72, 72, 72)',
                              }}
                            >
                              <path d="m23.25 24c-.19 0-.38-.07-.53-.22l-10.72-10.72-10.72 10.72c-.29.29-.77.29-1.06 0s-.29-.77 0-1.06l10.72-10.72-10.72-10.72c-.29-.29-.29-.77 0-1.06s.77-.29 1.06 0l10.72 10.72 10.72-10.72c.29-.29.77-.29 1.06 0s .29.77 0 1.06l-10.72 10.72 10.72 10.72c.29.29.29.77 0 1.06-.15.15-.34.22-.53.22" fillRule="evenodd" />
                            </svg>
                          </button>
                        </div>
                      </div>
                    </div>
                    <div className="full">
                      <div className="photoContainer">
                        <div className="photoDisplay">
                          <div>
                            <button type="button" name="Previous" className="leftButton" onClick={this.prevPictureHandler}>
                              <svg viewBox="0 0 18 18" name="Previous" role="presentation" aria-hidden="true" focusable="false" style={{ height: 24, width: 24, fill: 'rgb(72, 72, 72)' }}>
                                <path name="Previous" d="m13.7 16.29a1 1 0 1 1 -1.42 1.41l-8-8a1 1 0 0 1 0-1.41l8-8a1 1 0 1 1 1.42 1.41l-7.29 7.29z" fillRule="evenodd" />
                              </svg>
                            </button>
                            <button type="button" name="Next" className="rightButton" onClick={this.nextPictureHandler}>
                              <svg viewBox="0 0 18 18" name="Next" role="presentation" aria-hidden="true" focusable="false" style={{ height: 24, width: 24, fill: 'rgb(72, 72, 72)' }}>
                                <path name="Next" d="m4.29 1.71a1 1 0 1 1 1.42-1.41l8 8a1 1 0 0 1 0 1.41l-8 8a1 1 0 1 1 -1.42-1.41l7.29-7.29z" fillRule="evenodd" />
                              </svg>
                            </button>
                          </div>
                          <img className="hiddenPicture" aria-hidden="true" src={photos[currIndex].photoUrl} alt="" />
                          <span>
                            <div>
                              <img
                                src={photos[currIndex].photoUrl}
                                alt=""
                                onClick={this.nextPictureHandler}
                                data-veloute="slideshow-image"
                                style={{
                                  maxHeight: '55vh', objectFit: 'cover', cursor: 'pointer', zIndex: 2, borderRadius: 16, position: 'absolute', maxWidth: '85%', margin: '0px auto', top: 0, left: 0, right: 0, bottom: 0,
                                }}
                              />
                            </div>
                          </span>
                        </div>
                      </div>
                      <div className="miniCarouselContainer">
                        <div className="miniCarouselPad">
                          <div className="miniCarouselBottom">
                            <div className="miniCarouselOuter">
                              <div className="miniCarouselInner">
                                <div />
                                <div className="rightEdge" />
                                <div className="imageCenterer">
                                  <div className="_zf7q4">
                                    <ul className="_conlnu" style={{ transform: `translateX(${(translationAmount)}px)` }}>
                                      {photos.map(element => (
                                        element.priority === currIndex
                                          ? (
                                            <li className="_1y6wdia" key={element.priority}>
                                              <button data-photo-index="0" className="_1umg7png" aria-label={`1/${photos.length}: ${photos[0].caption}`}>
                                                <img alt="" src={photos[0].photoUrl} className="_u6tx3c" />
                                              </button>
                                            </li>
                                          )
                                          : (
                                            <li className="_12n0r6ug" key={element.priority}>
                                              <button data-photo-index={element.priority} className="_hwcst" aria-label={`${element.priority}/${photos.length}: ${photos[element.priority].caption}`}>
                                                <img alt="" src={photos[element.priority].photoUrl} className="_u6tx3c" />
                                              </button>
                                            </li>
                                          )
                                      ))}
                                    </ul>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                        <div className="_yoviva">
                          <div style={{ marginBottom: '16px' }}>
                            <div className="_1p3joamp">
                              {currIndex + 1}/{photos.length}
                            </div>
                          </div>
                          <div className="_1dcp06s">
                            <div className="_czm8crp">
                              {photos[currIndex].caption}
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Slideshow;
