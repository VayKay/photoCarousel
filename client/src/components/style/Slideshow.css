.modal {
    position: fixed !important;
    top: 0px !important;
    right: 0px !important;
    bottom: 0px !important;
    left: 0px !important;
    background: rgb(255, 255, 255) !important;
}

.slideshowModal {
    height: 100% !important;
    overflow-y: auto !important;
}


@media (min-width: 1128px) {
    .slideshowElements {
        display: table !important;
    }
}

.slideshowElements {
    width: 100% !important;
    height: 100% !important;
}

@media (min-width: 1128px) {
    .slideshowContainer {
        display: table-row !important;
    }
}

@media (min-width: 1440px) {
    .exitButton {
        margin-top: 64px !important;
        margin-right: 64px !important;
    }
}

@media (min-width: 744px) {
    .exitButton {
        right: 0px !important;
        margin-top: 40px !important;
        margin-right: 40px !important;
    }
}

.exitButton {
    position: absolute !important;
    z-index: 4 !important;
}

.hiddenButton {
    cursor: pointer !important;
    background-color: transparent !important;
    color: buttontext !important;
    border: 0px !important;
    display: block !important;
}

.hiddenButton:focus {
    outline: 0 !important;
}

svg:not(:root) {
    overflow: hidden;
}

@media (min-width: 1440px) {
    .photoContainer {
        padding: 64px !important;
    }
}

@media (min-width: 1128px) {
    .photoContainer {
        width: 65vw !important;
        display: table-cell !important;
        padding: 64px !important;
        margin-bottom: 0px !important;
        height: 75vh !important;
    }
}

@media (max-width: 1128px) {
    .photoContainer {
        margin-top: 20px !important;
        height: 50vh !important;
    }
}

@media (max-width: 744px) {
    .photoContainer {
        width: 100vw !important;
        height: 20vh !important;
    }
}

.photoContainer {
    vertical-align: middle !important;
}

@media (min-width: 1128px) {
    .photoDisplay {
        height: 100% !important;
    }
}

.photoDisplay {
    -webkit-box-pack: center !important;
    display: flex !important;
    position: relative !important;
    justify-content: center !important;
}

@media (min-width: 1128px) {
    .hiddenPicture {
        max-height: 75vh !important;
    }
}

.hiddenPicture {
    visibility: hidden !important;
    max-height: 55vh !important;
    max-width: 85% !important;
}

.leftButton {
    width: 50% !important;
    cursor: pointer !important;
    top: 0px !important;
    height: 100% !important;
    font-size: 0.5em !important;
    position: absolute !important;
    z-index: 1 !important;
    left: 0px !important;
    text-align: left !important;
    background: none !important;
    border-width: 0px !important;
    border-style: initial !important;
    border-color: initial !important;
    border-image: initial !important;
}

.rightButton {
    width: 50% !important;
    cursor: pointer !important;
    top: 0px !important;
    height: 100% !important;
    font-size: 0.5em !important;
    position: absolute !important;
    z-index: 1 !important;
    text-align: right !important;
    right: 0px !important;
    background: none !important;
    border-width: 0px !important;
    border-style: initial !important;
    border-color: initial !important;
    border-image: initial !important;
}

.leftButton:focus {
    outline: 0 !important;
}

.rightButton:focus {
    outline: 0 !important;
}

@media (min-width: 1440px) {
    .miniCarouselContainer {
        min-width: 384px !important;
        padding-top: 152px !important;
    }
}

@media (min-width: 1128px) {
    .miniCarouselContainer {
        padding-top: 152px !important;
        display: table-cell !important;
        vertical-align: top !important;
        width: 264px !important;
        min-width: 264px !important;
    }
}

@media (max-width: 1128px) {
    .miniCarouselContainer {
        /* margin-top: 170px !important; */
        padding-top: 100px !important;
    }
}

.miniCarouselContainer {
    overflow: hidden;
}

@media (min-width: 1128px) {
    .miniCarouselPad {
        width: 100% !important;
        padding-right: 52px !important;
        margin: 0px !important;
    }
}

.miniCarouselPad {
    margin-left: 40px !important;
    margin-right: 40px !important;
}

@media (min-width: 1440px) {
    .miniCarouselBottom {
        margin-bottom: 48px !important;
    }
}

@media (min-width: 1128px) {
    .miniCarouselBottom {
        width: 364px !important;
        margin-left: 0px !important;
        margin-right: 0px !important;
        margin-bottom: 32px !important;
    }
}

@media (max-width: 1127px) {
    .miniCarouselBottom {
        width: 50vw !important;
        margin: 0px auto 16px !important;
    }
}

.miniCarouselOuter {
    max-width: 105vh !important;
    margin-left: auto !important;
    margin-right: auto !important;
    overflow: hidden !important;
}

.miniCarouselInner {
    margin-left: auto !important;
    margin-right: auto !important;
    position: relative !important;
    transform: translateY(0px) !important;
    transition: -ms-transform 0.2s ease-out 0s, -webkit-transform 0.2s ease-out 0s, transform 0.2s ease-out 0s !important;
}

@media (min-width: 1440px) {
    .rightEdge {
        width: 32px !important;
        height: 96px !important;
    }
}

.rightEdge {
    position: absolute !important;
    right: 0px;
    height: 80px !important;
    z-index: 1 !important;
    width: 30px !important;
    background: linear-gradient(270deg, rgb(255, 255, 255) 0%, rgba(255, 255, 255, 0) 90.82%) !important;
}

@media (min-width: 1440px) {
    .leftEdge {
        width: 32px !important;
        height: 96px !important;
    }
}

.leftEdge {
    position: absolute !important;
    height: 80px !important;
    transform: rotate(180deg) !important;
    z-index: 1 !important;
    width: 30px !important;
    left:0px;
    background: linear-gradient(270deg, rgb(255, 255, 255) 0%, rgba(255, 255, 255, 0) 90.82%) !important;
    top: 0px;
}

@media (min-width: 1440px) {
    .imageCenterer {
        height: 96px !important;
        padding-top: 16px !important;
        padding-bottom: 16px !important;
    }
}

.imageCenterer {
    position: relative !important;
    padding-top: 8px !important;
    padding-bottom: 8px !important;
    height: 64px !important;
    overflow: hidden !important;
}


@media (min-width: 1440px) {
    .miniContainer {
        width: 8080px !important;
    }
}

.miniContainer {
    position: absolute !important;
    width: 5656px !important;
}

.translator {
    position: absolute !important;
    list-style-type: none !important;
    left: 0px !important;
    transition: -ms-transform 0.3s ease-out 0s, -webkit-transform 0.3s ease-out 0s, transform 0.3s ease-out 0s !important;
    margin: 0px !important;
    padding: 0px 0px 0px 8px !important;
}

@media (min-width: 1440px) {
    .miniFocus {
        margin-left: 16px !important;
    }
}

.miniFocus {
    float: left !important;
    background-color: rgb(255, 255, 255) !important;
    border-radius: 4px !important;
    margin-left: 16px !important;
}

@media (min-width: 1440px) {
    .miniFocusButton {
        width: 64px !important;
        height: 64px !important;
    }
}

.miniFocusButton {
    font-weight: inherit !important;
    backface-visibility: hidden !important;
    position: relative !important;
    display: inline-block !important;
    vertical-align: bottom !important;
    width: 64px !important;
    height: 64px !important;
    opacity: 1 !important;
    cursor: default !important;
    box-shadow: rgba(0, 0, 0, 0.16) 0px 0px 8px 2px, rgb(72, 72, 72) 0px 0px 0px 2px !important;
    overflow: hidden !important;
    background: transparent !important;
    border-width: 0px !important;
    border-style: initial !important;
    border-color: initial !important;
    border-image: initial !important;
    margin: 0px !important;
    padding: 0px !important;
    border-radius: 4px !important;
}

.miniImage {
    position: absolute !important;
    left: 50% !important;
    top: 50% !important;
    height: 100% !important;
    width: auto !important;
    transform: translate(-50%, -50%) !important;
    border-radius: 4px !important;
}

@media (min-width: 1440px) {
    .miniEntry {
        margin-left: 16px !important;
    }
}

.miniEntry {
    float: left !important;
    background-color: rgb(255, 255, 255) !important;
    margin-left: 16px !important;
    border-radius: 4px !important;
}

.miniEntry:focus {
    outline: none;
}

@media (min-width: 1440px) {
    .miniEntryButton {
        width: 64px !important;
        height: 64px !important;
    }
}

.miniEntryButton {
    font-weight: inherit !important;
    opacity: 0.7 !important;
    backface-visibility: hidden !important;
    position: relative !important;
    display: inline-block !important;
    vertical-align: bottom !important;
    width: 64px !important;
    height: 64px !important;
    overflow: hidden !important;
    background: transparent !important;
    border-width: 0px !important;
    border-style: initial !important;
    border-color: initial !important;
    border-image: initial !important;
    margin: 0px !important;
    padding: 0px !important;
    border-radius: 4px !important;
}

.miniEntryButton:hover {
    opacity: 1 !important;
}

@media (min-width: 1440px) {
    .descriptionContainer {
        padding-right: 64px !important;
    }
}

@media (min-width: 1128px) {
    .descriptionContainer {
        max-width: 100% !important;
        padding-right: 40px !important;
        margin: 0px !important;
    }
}

@media (min-width: 744px) {
    .descriptionContainer {
        max-width: 80% !important;
        margin: auto !important;
    }
}

.descriptionContainer {
    margin: 40px !important;
}

.indexCounter {
    margin: 0px !important;
    word-wrap: break-word !important;
    font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif !important;
    font-size: 16px !important;
    font-weight: 600 !important;
    line-height: 1.375em !important;
    color: #484848 !important;
}

.captionContainer {
    width: 350px;
    word-break: break-word !important;
}

.caption {
    margin: 0px !important;
    word-wrap: break-word !important;
    font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif !important;
    font-size: 16px !important;
    font-weight: 400 !important;
    line-height: 1.375em !important;
    color: #484848 !important;
}