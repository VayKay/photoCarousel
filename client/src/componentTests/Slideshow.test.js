/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { shallow } from 'enzyme';
import Slideshow from '../components/Slideshow';

describe('Slideshow', () => {
  it('should render correctly', () => {
    const component = shallow(<Slideshow
      modalFocus={1}
      photos={[{
        caption:
        'Et esse laudantium sunt et.',
        createdAt:
        '2019-07-10T01:42:08.000Z',
        id:
        5,
        listing_id:
        1,
        photoUrl:
        'http://lorempixel.com/1440/960/city',
        priority:
        0,
        updatedAt:
        '2019-07-10T01:42:08.000Z',
      },
      {
        caption:
        'Et esse laudantium sunt et.',
        createdAt:
        '2019-07-10T01:42:08.000Z',
        id:
        5,
        listing_id:
        1,
        photoUrl:
        'http://lorempixel.com/1440/960/city',
        priority:
        1,
        updatedAt:
        '2019-07-10T01:42:08.000Z',
      },
      ]}
    />);


    component.setState({ currIndex: 1, translationAmount: 8 });

    expect(component).toMatchSnapshot();
  });
});
