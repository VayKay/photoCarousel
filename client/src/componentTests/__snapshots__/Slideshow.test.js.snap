// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slideshow should render correctly 1`] = `
<div
  className="topmostSlideshow"
  style={
    Object {
      "visibility": undefined,
    }
  }
>
  <div>
    <div
      dir="ltr"
    >
      <div
        className="modal"
        role="dialog"
        style={
          Object {
            "zIndex": 2000,
          }
        }
      >
        <div
          className="slideshowModal"
          data-veloute="slideshow-modal"
        >
          <div
            className="slideshowElements"
          >
            <div
              className="slideshowContainer"
            >
              <div>
                <div>
                  <div
                    className="exitButton"
                  >
                    <button
                      aria-busy="false"
                      className="hiddenButton"
                      style={
                        Object {
                          "padding": 15,
                        }
                      }
                      type="button"
                    >
                      <svg
                        aria-label="Close"
                        focusable="false"
                        role="img"
                        style={
                          Object {
                            "display": "block",
                            "fill": "rgb(72, 72, 72)",
                            "height": 24,
                            "width": 24,
                          }
                        }
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="m23.25 24c-.19 0-.38-.07-.53-.22l-10.72-10.72-10.72 10.72c-.29.29-.77.29-1.06 0s-.29-.77 0-1.06l10.72-10.72-10.72-10.72c-.29-.29-.29-.77 0-1.06s.77-.29 1.06 0l10.72 10.72 10.72-10.72c.29-.29.77-.29 1.06 0s .29.77 0 1.06l-10.72 10.72 10.72 10.72c.29.29.29.77 0 1.06-.15.15-.34.22-.53.22"
                          fillRule="evenodd"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
              </div>
              <div
                className="full"
              >
                <div
                  className="photoContainer"
                >
                  <div
                    className="photoDisplay"
                  >
                    <div>
                      <button
                        className="leftButton"
                        name="Previous"
                        onClick={[Function]}
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          focusable="false"
                          name="Previous"
                          role="presentation"
                          style={
                            Object {
                              "fill": "rgb(72, 72, 72)",
                              "height": 24,
                              "width": 24,
                            }
                          }
                          viewBox="0 0 18 18"
                        >
                          <path
                            d="m13.7 16.29a1 1 0 1 1 -1.42 1.41l-8-8a1 1 0 0 1 0-1.41l8-8a1 1 0 1 1 1.42 1.41l-7.29 7.29z"
                            fillRule="evenodd"
                            name="Previous"
                          />
                        </svg>
                      </button>
                      <button
                        className="rightButton"
                        name="Next"
                        onClick={[Function]}
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          focusable="false"
                          name="Next"
                          role="presentation"
                          style={
                            Object {
                              "fill": "rgb(72, 72, 72)",
                              "height": 24,
                              "width": 24,
                            }
                          }
                          viewBox="0 0 18 18"
                        >
                          <path
                            d="m4.29 1.71a1 1 0 1 1 1.42-1.41l8 8a1 1 0 0 1 0 1.41l-8 8a1 1 0 1 1 -1.42-1.41l7.29-7.29z"
                            fillRule="evenodd"
                            name="Next"
                          />
                        </svg>
                      </button>
                    </div>
                    <img
                      alt=""
                      aria-hidden="true"
                      className="hiddenPicture"
                      src="http://lorempixel.com/1440/960/city"
                    />
                    <div>
                      <img
                        alt=""
                        data-veloute="slideshow-image"
                        id="mainPicture"
                        onClick={[Function]}
                        src="http://lorempixel.com/1440/960/city"
                        style={
                          Object {
                            "borderRadius": 16,
                            "bottom": 0,
                            "cursor": "pointer",
                            "left": 0,
                            "margin": "0px auto",
                            "maxHeight": "75vh",
                            "maxWidth": "85%",
                            "objectFit": "cover",
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                            "zIndex": 2,
                          }
                        }
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="miniCarouselContainer"
                >
                  <div
                    className="miniCarouselPad"
                  >
                    <div
                      className="miniCarouselBottom"
                    >
                      <div
                        className="miniCarouselOuter"
                      >
                        <div
                          className="miniCarouselInner"
                        >
                          <div />
                          <div
                            className="leftEdge"
                          />
                          <div
                            className="rightEdge"
                          />
                          <div
                            className="imageCenterer"
                          >
                            <div
                              className="miniContainer"
                            >
                              <ul
                                className="translator"
                                style={
                                  Object {
                                    "transform": "translateX(8px)",
                                  }
                                }
                              >
                                <li
                                  className="miniEntry"
                                  key="0"
                                  onClick={[Function]}
                                >
                                  <button
                                    aria-label="0/2: Et esse laudantium sunt et."
                                    className="miniEntryButton"
                                    index={0}
                                  >
                                    <img
                                      alt=""
                                      className="miniImage"
                                      index={0}
                                      src="http://lorempixel.com/1440/960/city"
                                    />
                                  </button>
                                </li>
                                <li
                                  className="miniFocus"
                                  key="1"
                                >
                                  <button
                                    aria-label="1/2: Et esse laudantium sunt et."
                                    className="miniFocusButton"
                                    index="0"
                                  >
                                    <img
                                      alt=""
                                      className="miniImage"
                                      src="http://lorempixel.com/1440/960/city"
                                    />
                                  </button>
                                </li>
                              </ul>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    className="descriptionContainer"
                  >
                    <div
                      style={
                        Object {
                          "marginBottom": "16px",
                        }
                      }
                    >
                      <div
                        className="indexCounter"
                      >
                        2
                        /
                        2
                      </div>
                    </div>
                    <div
                      className="captionContainer"
                    >
                      <div
                        className="caption"
                      >
                        Et esse laudantium sunt et.
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
